### PR TITLE
Add tutorials for V2.0

### PR DESCRIPTION
**Pull request by ZwwWayne**
This PR refactors the docs and add tutorials to meet user's needs, such as in https://github.com/open-mmlab/mmdetection/issues/2424, https://github.com/open-mmlab/mmdetection/issues/2314, https://github.com/open-mmlab/mmdetection/issues/1837, https://github.com/open-mmlab/mmdetection/issues/1648. 
It provides three tutorials:

1.  Tutorial for adding new components including backbone, heads, and loss.
2. Tutorial for adding a new dataset and mixing (concatenate/repeat) several datasets.
3. Tutorial for fine-tuning in downstream tasks using pre-trained models.

### CODE REVIEW

File: docs/tutorials/finetune.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Tutorial 3: Finetuning Models
+
+Detectors pre-trained on the COCO dataset always serve as a good pre-trained model for other datasets, e.g., CityScapes and KITTI Dataset.
```
Comment (hellock) [inline docs/tutorials/finetune.md:3]: always -> can

File: docs/tutorials/new_dataset.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
+# Tutorials 2: Adding New Dataset
```
Comment (hellock) [inline docs/tutorials/new_dataset.md:1]: We also need to tell how to modify class names with existing dataset types.

File: docs/tutorials/finetune.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            num_classes=8,
+            loss_mask=dict(
+                type='CrossEntropyLoss', use_mask=True, loss_weight=1.0))))
+```
+
+## Modify dataset
+The users may also need to prepare the dataset and write the configs about dataset. MMDetection V2.0 already support VOC, WIDER FACE, COCO and Cityscapes Dataset.
+
+## Modify training schedule
+The finetunings hyperparameters varies from the default schedule. It usually requires smaller learning and less training epochs
```
Comment (hellock) [inline docs/tutorials/finetune.md:57]: finetunings -> finetuning
varies -> vary
learning -> learning rate

File: docs/tutorials/new_dataset.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+```python
+dataset_A_train = dict(
+        type='RepeatDataset',
+        times=N,
+        dataset=dict(
+            type='Dataset_A',
+            ...
+            pipeline=train_pipeline
+        )
+    )
```
Comment (hellock) [inline docs/tutorials/new_dataset.md:188]: Note the indents.

File: docs/tutorials/new_dataset.md [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+        dataset_A_train,
+        dataset_B_train
+    ],
+    val = dataset_A_val,
+    test = dataset_A_test
+    )
+
+```
+
+### Modify classes of existing dataset
```
Comment (hellock) [inline docs/tutorials/new_dataset.md:219]: Another common case is that users have converted custom datasets to existing format, e.g., COCO, VOC, but the classes are different. We need to provide examples to show how to overwrite existing classes, not necessarily being a subset. The classes can be either a tuple of class names or a filename.

File: docs/tutorials/new_modules.md [hunk 6]
```diff
@@ -0,0 +1,321 @@
+# Tutorials 1: Adding New Modules
+
+## Customize optimizer
```
Comment (hellock) [inline docs/tutorials/new_modules.md:3]: Also provide an example of customizing optimizer constructors.

File: docs/tutorials/new_modules.md [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+Then the users need to add it in the `mmdet/models/losses/__init__.py`.
+```python
+from .my_loss import MyLoss, my_loss
+
+```
+
+To use it, modify the `loss_xxx` field.
+Since MyLoss is for regrression, you need to modify the `loss_bbox` field in the head.
+```python
+loss_bbox=dict(type='L1Loss', loss_weight=1.0))
```
Comment (hellock) [inline docs/tutorials/new_modules.md:320]: L1Loss -> MyLoss
